File: flutter/complex_app/lib/services/mock_book_service.dart
import '../models/book.dart';

class MockBookService {
  static final List<Book> _books = List.generate(
    10000,
    (i) => Book(
      id: '${i + 1}',
      title: 'Book Title ${i + 1}',
      author: 'Author ${i % 100 + 1}',
      description: 'Description for Book ${i + 1}',
    ),
  );

  static List<Book> getBooks() => _books;
  static Book? getBookById(String id) =>
      _books.firstWhere((b) => b.id == id, orElse: () => _books[0]);
}

File: kotlin_multiplatform/complex_app/composeApp/src/commonMain/kotlin/com/example/complex_app/services/MockBookService.kt
package com.example.complex_app.services

import com.example.complex_app.models.Book

object MockBookService {
    private val books = List(10000) {
        Book(
            id = it.toString(),
            title = "Book $it",
            author = "Author ${it % 100}",
            description = "Description for Book $it",
            userId = (it % 100).toString()
        )
    }
    fun getBooks(query: String? = null): List<Book> =
        books.filter { query == null || it.title.contains(query, true) || it.author.contains(query, true) }
    fun getBook(id: String): Book? = books.find { it.id == id }
    fun getBooksByUser(userId: String): List<Book> = books.filter { it.userId == userId }
}
